lib/mbta_v3_api/repository.ex

```diff
@@ -60,28 +60,32 @@ defmodule MBTAV3API.Repository.Impl do
 
   use Nebulex.Caching.Decorators
 
-  alias MBTAV3API.JsonApi
+  alias MBTAV3API.{JsonApi, RepositoryCache}
 
   @ttl :timer.hours(1)
 
   @impl true
+  @decorate cacheable(cache: RepositoryCache, on_error: :nothing, opts: [ttl: @ttl])
   def route_patterns(params, opts \\ []), do: all(MBTAV3API.RoutePattern, params, opts)
 
   @impl true
+  @decorate cacheable(cache: RepositoryCache, on_error: :nothing, opts: [ttl: @ttl])
   def routes(params, opts \\ []), do: all(MBTAV3API.Route, params, opts)
 
   @impl true
+  @decorate cacheable(cache: RepositoryCache, on_error: :nothing, opts: [ttl: @ttl])
   def schedules(params, opts \\ []), do: all(MBTAV3API.Schedule, params, opts)
 
   @impl true
+  @decorate cacheable(cache: RepositoryCache, on_error: :nothing, opts: [ttl: @ttl])
   def stops(params, opts \\ []), do: all(MBTAV3API.Stop, params, opts)
 
   @impl true
+  @decorate cacheable(cache: RepositoryCache, on_error: :nothing, opts: [ttl: @ttl])
   def trips(params, opts \\ []), do: all(MBTAV3API.Trip, params, opts)
 
   @spec all(module(), JsonApi.Params.t(), Keyword.t()) ::
           {:ok, JsonApi.Response.t(JsonApi.Object.t())} | {:error, term()}
-  @decorate cacheable(cache: MBTAV3API.RepositoryCache, on_error: :nothing, opts: [ttl: @ttl])
   defp all(module, params, opts) do
```

Comment:
Since the first argument here is different for every type of query being made, I would expect that keeping the decorator on `all/3` would be fine - perhaps since there are no default arguments at play that would let us do something like
```elixir
@impl Nebulex.Caching.KeyGenerator
def generate(_, _, [module, params, opts]) do
  "#{module}|#{:erlang.phash2({params, opts})}"
end
```
Maybe that's too clever, though, and it still won't work and we'll be confused trying to debug it.

Reply:
Yeah that does seem like it would be equivalent. I do think it is beneficial to have the decorator on each function name so it is clearer when looking just at the repository module that caching is done at the function-level without having to dig into the key generation